src/python/dxpy/workflow_builder.py

```diff
@@ -69,6 +69,32 @@ def _parse_executable_spec(src_dir, json_file_name, parser):
             raise WorkflowBuilderException("Could not parse {} file as JSON: {}".format(json_file_name, e.args))
 
 
+def _fetch_spec_from_workflow(args, parser):
+    try:
+        source_workflow = dxpy.DXWorkflow(args._from)
+        json_spec = source_workflow.describe(fields={"properties", "details"}, default_fields=True)
+        json_spec = _cleanup_empty_keys(json_spec)
+
+    except Exception as e:
+        raise WorkflowBuilderException("Could not get specs from given workflow {}: {}".format(args._from, e.args))
+
+    return json_spec
+
+def _cleanup_empty_keys(json_spec):
+    import re
+    clean_json = re.sub('\"\w*\": (null|\{\}|\"\"|\[\])(\,|)\s*','',json.dumps(json_spec)).replace(", }","}")
+    return json.loads(clean_json)
+
+def _check_dxcompiler_version(json_spec):
+    if  json_spec.get("details") and json_spec["details"].get("version"):
+        from distutils.version import StrictVersion
+        supported_version = "2.5.0"
```

Comment:
Before merging, this should be updated to the actual dxCompiler version where the feature is supported (which we don't know yet until dxCompiler part is done).